src/controllers/userController.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-else-return */
+/* eslint-disable keyword-spacing */
+/* eslint-disable no-unused-vars */
+/* eslint-disable require-jsdoc */
+import sgMail from '@sendgrid/mail';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+import Util from '../utils/Utils';
+import Mail from '../utils/Mail';
+import userService from '../services/UserServices';
+import AuthenticationHelper from '../utils/AuthHelper';
+
+const util = new Util();
+const mail = new Mail();
+const { jwtSignReset, jwtVerify } = AuthenticationHelper;
+class UserController {
+  static async reset(req, res) {
+    // check whether email address is recorded
+    const searchUser = await userService.findUserByEmail(req.body.email);
+    if (!searchUser) {
+      util.setError(409, 'Email not found');
+      return util.send(res);
+    }
+    // sign token
+    const token = jwtSignReset(req.body.email);
+    // store password reset
+    const userinfo = {
+      user_id: searchUser.id,
+      token
+    };
+    const savetoken = await userService.storeToken(userinfo).then(() => {
+      // send email
+      mail.setMessage(req.body.email, token, searchUser.firstname, req.body.email);
+      return mail.send(res);
+    });
+  }
+
+  static async updateCredentials(req, res) {
+    // check if password looks alike
+    if (req.body.password === req.body.confirm_password) {
```

Comment:
I think this should be a comparison of the password hashes with Bcrypt compare

Reply:
@didasmbalanya That condition is used to compare the inputs from the user, just to check if the password he inputed is the same, so i don't really understand the usage of Bcrypt here, If you can explain to me better what you mean, that would be great

Reply:
oh it's just confirmed password? okay I thought it was comparing it to encrypted password from the database. It's okay, cool